### PR TITLE
[Enhancement] Get table binding of builtin storage volume

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 21 (95.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/SharedNothingStorageVolumeMgr.java|0|1|00.00%|[163]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java [hunk 1]
```diff
@@ -306,6 +308,25 @@ public void validateStorageVolumeConfig() throws InvalidConfException {
...
+            Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+            db.readLock();
+            List<Table> tables = db.getTables();
+            for (Table table : tables) {
+                Long tableId = table.getId();
+                if (!tableToStorageVolume.containsKey(tableId) && table.isCloudNativeTableOrMaterializedView()) {
+                    tableIds.add(tableId);
+                }
+            }
+            db.readUnlock();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java:325]:
```
db.readLock();
try {

} finally {
    db.readUnlock();
}
```

File: fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java [hunk 2]
```diff
@@ -306,6 +308,25 @@ public void validateStorageVolumeConfig() throws InvalidConfException {
         }
     }
 
+    @Override
+    protected Set<Long> getTableBindingsOfBuiltinStorageVolume() {
+        Set<Long> tableIds = new HashSet<>();
+        List<Long> dbIds = GlobalStateMgr.getCurrentState().getDbIds();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java:314]: should also check the dbs and tables in `CatalogRecycleBin`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/SharedNothingStorageVolumeMgr.java|0|1|00.00%|[163]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|6|6|100.00%|[]|

**Review by kevincai — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java [hunk 3]
```diff
@@ -306,6 +308,25 @@ public void validateStorageVolumeConfig() throws InvalidConfException {
...
+        for (Long dbId : dbIds) {
+            Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+            db.readLock();
+            List<Table> tables = db.getTables();
+            for (Table table : tables) {
+                Long tableId = table.getId();
+                if (!tableToStorageVolume.containsKey(tableId) && table.isCloudNativeTableOrMaterializedView()) {
```
[the remark this answers is not in the document]
Comment (TszKitLo40) [reply fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java]: write lock has been used in `removeStorageVolume`.

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 21 (95.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/SharedNothingStorageVolumeMgr.java|0|1|00.00%|[163]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|6|6|100.00%|[]|

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**